poe-api/src/poe.py

```diff
@@ -147,6 +147,16 @@ def get_bot_names(self):
       bot_names[bot_nickname] = bot_obj["displayName"]
     return bot_names
   
+  def get_remaining_messages(self, chatbot):
+    url = f'https://poe.com/_next/data/{self.next_data["buildId"]}/{chatbot}.json'
+    limited_chatbots = ["GPT-4", "Claude+"]
+    if chatbot in limited_chatbots:
+      r = request_with_retries(self.session.get, url)
+      remaining_count = r.json()["pageProps"]["payload"]["chatOfBotDisplayName"]["defaultBotObject"]["messageLimit"]["numMessagesRemaining"]
+      return remaining_count
+    else:
+      raise Exception(f"Chatbot {chatbot} is not limited or doesn't exist.")
```

Comment:
Maybe don't raise an exception for if the chatbot doesn't have a limit. The `numMessagesRemaining` value is present on all chatbots anyways, and it'll just be `null` if there isn't a limit.